cpp/modmesh/toggle/profile.hpp

```diff
@@ -156,6 +166,39 @@ class TimeRegistry
         return ostm.str();
     }
 
+    std::string detailed_report() const
```

Comment:
This function body is long.  If we do not give it an implementation file (`.cpp`), it should be taken outside the class declaration.

(If you are not sure what does it mean, see https://github.com/solvcon/modmesh/pull/377#discussion_r1652421400 .)